packages/sync/src/events.ts

```diff
@@ -0,0 +1,61 @@
+import { DidDocument } from '@atproto/identity'
+import type { RepoRecord } from '@atproto/lexicon'
+import { BlockMap } from '@atproto/repo'
+import { AtUri } from '@atproto/syntax'
+import type { CID } from 'multiformats/cid'
+
+export type Event = CommitEvt | IdentityEvt | AccountEvt
+
+export type CommitMeta = {
+  seq: number
+  time: string
+  commit: CID
+  blocks: BlockMap
+  rev: string
+  uri: AtUri
+  did: string
+  collection: string
+  rkey: string
+}
+
+export type CommitEvt = Create | Update | Delete
+
+export type Create = CommitMeta & {
+  event: 'create'
+  record: RepoRecord
+  cid: CID
+}
+
+export type Update = CommitMeta & {
+  event: 'update'
+  record: RepoRecord
+  cid: CID
+}
+
+export type Delete = CommitMeta & {
+  event: 'delete'
+}
+
+export type IdentityEvt = {
+  seq: number
+  time: string
+  event: 'identity'
+  did: string
+  handle?: string
+  didDocument: DidDocument | null
```

Comment:
Is there room to standardize around optional fields or `null` fields?  Or perhaps the semantics between the two are intentionally different here.

Reply:
_kinda_ different semantics, but not enough.  i'll make them both `undefined` 👌